### PR TITLE
Download data from Volumes

### PR DESCRIPTION
**Pull request by NikitaYurasov**
Add support to download data from Volumes (managed and external)

Minor features:
- future results have `.df()` method to get pandas DataFrame

### CODE REVIEW

File: dbxio/delta/table_commands.py [hunk 1]
```diff
@@ -225,12 +226,14 @@ def bulk_write_table(
     pa_table = create_pa_table(columnar_table, schema=dbxio_table.schema)
     pa_table_as_bytes = pa_table2parquet(pa_table)
 
-    abs_client = get_blob_servie_client(
-        abs_name,
-        az_cred_provider=client.credential_provider.az_cred_provider,
-    ).get_container_client(abs_container_name)
+    object_storage = ObjectStorageClient.from_storage_options(
+        scheme='abfss',  # TODO: make cloud-agnostic
```
Comment (solesensei) [inline dbxio/delta/table_commands.py:230]: Can we move this to `Settings()`? To setup schemes at least with env vars.

File: dbxio/utils/object_storage/_aws.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+from dbxio.utils.object_storage.object_storage import ObjectStorageClient
+
+
+@attrs.define
+class _S3StorageClientImpl(ObjectStorageClient):
+    bucket_name: str = attrs.field(validator=attrs.validators.instance_of(str))
+    object_key: str = attrs.field(validator=attrs.validators.instance_of(str))
+    domain_name: Optional[str] = None
+
+    # scheme: str = attrs.field(default='s3', init=False)
```
Comment (solesensei) [inline dbxio/utils/object_storage/_aws.py:15]: why comment?

File: dbxio/volume/volume_commands.py [hunk 3]
```diff
@@ -82,6 +95,75 @@ def _exists_volume(volume: Volume, client: 'DbxIOClient') -> bool:
...
+    Volume must be an external volume.
+
+    :param path: local directory to download files to. Must be a directory.
+    :param catalog_name: databricks catalog name
+    :param schema_name: databricks schema name
+    :param volume_name: volume name
+    :param client: dbxio client
+    """
+    path = Path(path)
+    assert path.is_dir(), 'Path must be a directory'
```
Comment (solesensei) [inline dbxio/volume/volume_commands.py:147]: It's better to use ValueError here

File: dbxio/delta/table_commands.py (hunk 1 above)
Comment (NikitaYurasov) [reply dbxio/delta/table_commands.py]: Moved it to the `Settings`

File: dbxio/core/cloud/__init__.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+from dbxio.core.cloud.gcp import GCP_ClOUD_PROVIDER_NAME
+from dbxio.core.cloud.nebius_over_azure import NEBIUS_OVER_AZURE_ClOUD_PROVIDER_NAME
+
+
+@enum.unique
+class CloudProvider(str, enum.Enum):
+    AWS = AWS_ClOUD_PROVIDER_NAME
+    AZURE = AZURE_ClOUD_PROVIDER_NAME
+    GCP = GCP_ClOUD_PROVIDER_NAME
+    NEBIUS_OVER_AZURE = NEBIUS_OVER_AZURE_ClOUD_PROVIDER_NAME
```
Comment (solesensei) [inline dbxio/core/cloud/__init__.py:15]: What is nebius_over_azure? Why not just nebius?

File: dbxio/core/settings.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+CLOUD_PROVIDER_ENV_VAR = 'CLOUD_PROVIDER'
+
+
+def _cloud_provider_factory() -> CloudProvider:
+    return CloudProvider(os.getenv(CLOUD_PROVIDER_ENV_VAR, _DEFAULT_CLOUD_PROVIDER).lower())
+
+
+@attrs.define
+class Settings:
+    cloud_provider: CloudProvider = attrs.field(factory=_cloud_provider_factory)
```
Comment (solesensei) [inline dbxio/core/settings.py:19]: I thought about some kind of _global_ settings like `pydantic_settings`. But this is ok

**Review by solesensei — DISMISSED** [review]

File: dbxio/core/cloud/__init__.py (hunk 4 above)
Comment (NikitaYurasov) [reply dbxio/core/cloud/__init__.py:15]: Because Nebius has it's own authorization system, but we use here azure on nebius machines

Comment (NikitaYurasov) [reply dbxio/core/cloud/__init__.py]: Removed Nebius as cloud provider

**Pull request merged**